compositor_render/src/renderer/color_converter_pipeline.rs

```diff
@@ -234,3 +236,91 @@ impl RGBAToYUVConverter {
         }
     }
 }
+
+pub struct R8FillWithValuePipeline {
+    pipeline: wgpu::RenderPipeline,
+    buffers: RectangleRenderBuffers,
+}
+
+impl R8FillWithValuePipeline {
+    pub fn new(device: &wgpu::Device) -> Self {
+        let shader_module = device.create_shader_module(wgpu::include_wgsl!("r8_fill_color.wgsl"));
+        let buffers = RectangleRenderBuffers::new(device);
+
+        let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+            label: Some("Fill with color render pipeline layout"),
+            bind_group_layouts: &[],
+            push_constant_ranges: &[wgpu::PushConstantRange {
+                stages: wgpu::ShaderStages::FRAGMENT,
+                range: 0..4,
+            }],
+        });
+
+        let pipeline = device.create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+            label: Some("Fill with color pipeline"),
+            layout: Some(&pipeline_layout),
+            primitive: PRIMITIVE_STATE,
+            vertex: wgpu::VertexState {
+                module: &shader_module,
+                entry_point: "vs_main",
+                buffers: &[Vertex::LAYOUT],
+            },
+            fragment: Some(wgpu::FragmentState {
+                module: &shader_module,
+                entry_point: "fs_main",
+                targets: &[Some(wgpu::ColorTargetState {
+                    format: wgpu::TextureFormat::R8Unorm,
+                    write_mask: wgpu::ColorWrites::all(),
+                    blend: None,
+                })],
+            }),
+            depth_stencil: None,
+            multisample: wgpu::MultisampleState {
+                count: 1,
+                mask: !0,
+                alpha_to_coverage_enabled: false,
+            },
+            multiview: None,
+        });
+
+        Self { pipeline, buffers }
+    }
+
+    pub fn fill(&self, ctx: &WgpuCtx, dst: &Texture, value: f32) {
+        let mut encoder = ctx
+            .device
+            .create_command_encoder(&wgpu::CommandEncoderDescriptor {
+                label: Some("Fill R8 texture with color command encoder"),
+            });
+
+        {
+            let mut render_pass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                label: Some("Fill R8 texture with color render pass"),
+                color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                    ops: wgpu::Operations {
+                        load: wgpu::LoadOp::Clear(wgpu::Color::BLACK),
```

Comment:
Just use this

Reply:
https://docs.rs/wgpu/latest/wgpu/enum.LoadOp.html

Reply:
Passing `fill color` as https://docs.rs/wgpu/latest/wgpu/struct.Color.html  instead of `wgpu::Color::BLACK` does the whole job :)

Reply:
responded on slack